cpp/cuproj/benchmarks/fixture/rmm_pool_raii.hpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <rmm/mr/device/cuda_memory_resource.hpp>
+#include <rmm/mr/device/owning_wrapper.hpp>
+#include <rmm/mr/device/per_device_resource.hpp>
+#include <rmm/mr/device/pool_memory_resource.hpp>
+
+namespace cuspatial {
+
+/**
+ * @brief An RAII class setting up RMM memory pool for `nvbench` benchmarks
+ *
+ * This is a temporary solution before templated fixtures tests are supported
+ * in `nvbench`. Similarly to `cuspatial::benchmark`, creating this RAII object in
+ * each benchmark will ensure that the RAPIDS Memory Manager pool mode is used
+ * in benchmarks, which eliminates memory allocation / deallocation performance
+ * overhead from the benchmark.
+ *
+ * Example:
+ *
+ * void my_benchmark(nvbench::state& state) {
+ * cuspatial::rmm_pool_raii pool_raii;
+ * state.exec([](nvbench::launch& launch) {
+ *       // benchmark stuff
+ *  });
+ * }
+ *
+ * NVBENCH_BENCH(my_benchmark);
+ */
+class rmm_pool_raii {
```

Comment:
This class seems totally unused.

Reply:
Removing. This was for when I was using nvbench.